src/Simulation/CsharpGeneration.Tests/SimulationCodeTests.fs

```diff
@@ -3204,3 +3203,112 @@ namespace Microsoft.Quantum.Tests.LineNumbers
+[assembly: SpecializationDeclaration("{\"Kind\":{\"Case\":\"QsBody\"},\"TypeArguments\":{\"Case\":\"Null\"},\"Information\":{\"Characteristics\":{\"Case\":\"EmptySet\"},\"InferredInformation\":{\"IsSelfAdjoint\":false,\"IsIntrinsic\":false}},\"Parent\":{\"Namespace\":\"Microsoft.Quantum.Tests.UnitTests\",\"Name\":\"UnitTest1\"},\"Attributes\":[],\"SourceFile\":\"%%%\",\"Position\":{\"Item1\":20,\"Item2\":4},\"HeaderRange\":{\"Item1\":{\"Line\":1,\"Column\":11},\"Item2\":{\"Line\":1,\"Column\":20}},\"Documentation\":[]}")]
+
+#line hidden
+namespace Microsoft.Quantum.Core
+{
+    public class Attribute : UDTBase<QVoid>, IApplyData
+    {
+        public Attribute() : base(default(QVoid))
+        {
+        }
+
+        public Attribute(QVoid data) : base(data)
+        {
+        }
+
+        System.Collections.Generic.IEnumerable<Qubit> IApplyData.Qubits => null;
+        public void Deconstruct()
+        {
+        }
+    }
+}
+
+#line hidden
+namespace Microsoft.Quantum.Diagnostics
+{
+    public class TestOperation : UDTBase<String>, IApplyData
+    {
+        public TestOperation() : base(default(String))
+        {
+        }
+
+        public TestOperation(String data) : base(data)
+        {
+        }
+
+        System.Collections.Generic.IEnumerable<Qubit> IApplyData.Qubits => null;
+        public void Deconstruct()
+        {
+        }
+    }
+}
+
+#line hidden
+namespace Microsoft.Quantum.Tests.UnitTests
+{
+    [SourceLocation("%%%", OperationFunctor.Body, 21, -1)]
+    public partial class UnitTest1 : Operation<QVoid, QVoid>, ICallable
+    {
+        public UnitTest1(IOperationFactory m) : base(m)
+        {
+        }
+
+        String ICallable.Name => "UnitTest1";
+        String ICallable.FullName => "Microsoft.Quantum.Tests.UnitTests.UnitTest1";
+        public override Func<QVoid, QVoid> Body => (__in__) =>
+        {
+#line hidden
+            return QVoid.Instance;
+        }
+
+        ;
+        public override void Init()
+        {
+        }
+
+        public override IApplyData __dataIn(QVoid data) => data;
+        public override IApplyData __dataOut(QVoid data) => data;
+        public static System.Threading.Tasks.Task<QVoid> Run(IOperationFactory __m__)
+        {
+            return __m__.Run<UnitTest1, QVoid, QVoid>(QVoid.Instance);
+        }
+    }
+
+    public partial class QuantumSimulatorTests
```

Comment:
I wouldn't name this `QuantumSimulatorTests`. It can clash, but also, these are not tests for the quantum simulator, so the name is missleading.
What about something like simply `__AllTests__`.

Reply:
Would this involve a rename of the partial class definitions for QuantumSimulatorTests found in BasicTests.cs, CircuitsTests.cs, and VerifyGates.cs under Simulators.Tests\QuantumSimulatorTests?

Reply:
I think the bundling according to what target they run on is both useful and actually also needed. How about something like "__ExecutedOnQuantumSimulator__" instead?